test/ref/scatter_elements.cpp

```diff
@@ -0,0 +1,392 @@
+
+    migraphx::shape su{migraphx::shape::float_type, {1, 2}};
+    std::vector<float> vu = {1.1, 2.1};
+
+    auto ld = mm->add_literal(migraphx::literal{sd, vd});
+    auto li = mm->add_literal(migraphx::literal{si, vi});
+    auto lu = mm->add_literal(migraphx::literal{su, vu});
+    auto r  = mm->add_instruction(
+        migraphx::make_op("scatter_elements", {{"axis", axis}, {"reduction", reduction_mode}}),
+        ld,
+        li,
+        lu);
+    mm->add_return({r});
+    return p;
+}
+
+TEST_CASE(scatter_elements_none_axis_1_test)
+{
+    migraphx::program p = create_scatter_elements_program2("none", 1);
+    p.compile(migraphx::make_target("ref"));
+
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<float> gold_none = {1.0, 1.1, 3.0, 2.1, 5.0};
+    EXPECT(migraphx::verify::verify_rms_range(results_vector, gold_none));
+}
+
+TEST_CASE(scatter_elements_mul_axis_1_test)
+{
+    migraphx::program p = create_scatter_elements_program2("mul", 1);
+    p.compile(migraphx::make_target("ref"));
+
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<float> gold_mul = {1.0, 2.2, 3.0, 8.4, 5.0};
+    EXPECT(migraphx::verify::verify_rms_range(results_vector, gold_mul));
+}
+
+TEST_CASE(scatter_elements_add_axis_1_test)
+{
+    migraphx::program p = create_scatter_elements_program2("add", 1);
+    p.compile(migraphx::make_target("ref"));
+
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<float> gold_add = {1.0, 3.1, 3.0, 6.1, 5.0};
+    EXPECT(migraphx::verify::verify_rms_range(results_vector, gold_add));
+}
+
+TEST_CASE(scatter_elements_min_axis_1_test)
+{
+    migraphx::program p = create_scatter_elements_program2("min", 1);
+    p.compile(migraphx::make_target("ref"));
+
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<float> gold_min = {1.0, 1.1, 3.0, 2.1, 5.0};
+    EXPECT(migraphx::verify::verify_rms_range(results_vector, gold_min));
+}
+
+TEST_CASE(scatter_elements_max_axis_1_test)
+{
+    migraphx::program p = create_scatter_elements_program2("max", 1);
+    p.compile(migraphx::make_target("ref"));
+
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+
+    std::vector<float> gold_max = {1.0, 2.0, 3.0, 4.0, 5.0};
+    EXPECT(migraphx::verify::verify_rms_range(results_vector, gold_max));
+}
+
+auto scatter_elements_duplicate_index_test(const std::string& reduction_mode)
```

Comment:
I don't see anything in the code to check for duplicate index with reduction mode "none," which is an error.

Reply:
The ONNX spec does state the following "In cases where reduction is set to "none", indices should not have duplicate entries: that is, if idx1 != idx2, then indices[idx1] != indices[idx2]", but I'm not sure that warrants checking for and throwing an error because of the overhead it adds. I'm of the opinion that it should be left as is.